onadata/apps/main/views.py

```diff
@@ -1445,3 +1446,32 @@ def get_context_data(self, **kwargs):
         context['user'] = self.request.user
         context['request_path'] = self.request.get_full_path()
         return context
+
+
+class CustomLoginView(LoginView):
+    """
+    Custom LoginView subclass
+    """
+
+    def get_template_names(self, **kwargs):
+        """
+        get custom login template name if use_custom_login_template is true
+        """
+        context = super(CustomLoginView, self).get_context_data(**kwargs)
+        if settings.USE_CUSTOM_LOGIN_TEMPLATE:
+            context['CUSTOM_LOGIN_TEMPLATE_PAGE'] = \
+                settings.CUSTOM_LOGIN_TEMPLATE_PAGE
+            return context['CUSTOM_LOGIN_TEMPLATE_PAGE']
```

Comment:
why not do `return [settings.CUSTOM_LOGIN_TEMPLATE_PAGE]`?  You don't need to mess with context here, in this method.